src/libmongoc/src/mongoc/mongoc-cluster-aws.c

```diff
@@ -967,6 +1079,9 @@ _mongoc_cluster_auth_node_aws (mongoc_cluster_t *cluster,
 
    ret = true;
 fail:
+   if (!ret) {
+      _mongoc_aws_credentials_cache_clear ();
```

Comment:
Can/should clearing of the credentials cache be moved into `_mongoc_aws_credentials_obtain` with `_mongoc_aws_credentials_cache_clear_nolock`? It is odd that the cache is cleared in a separate lock from the operations that determine the "should clear cache" condition.